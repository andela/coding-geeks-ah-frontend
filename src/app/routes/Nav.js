import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../../logo.png';

const Navbar = () => (
  <nav className="nav-wrapper">
    <Link to="/">
      <img src={logo} className="App-logo" alt="authors haven logo" />
    </Link>
    <ul className="right">
      <li>
        <Link to="/">Home</Link>
      </li>
      <li>
        <Link to="/About" className="last">About</Link>
      </li>
    </ul>
  </nav>
);

export default Navbar;
