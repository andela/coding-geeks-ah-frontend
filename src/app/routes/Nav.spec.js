import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { mount } from 'enzyme';
import sinon from 'sinon';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Nav from './Nav';

const middlewares = [thunk, promiseMiddleware];

let url = null;
const mockStore = configureStore(middlewares);
const store = mockStore({
  isAuthenticated: true,
  social: jest.fn(),
  login: jest.fn(),
});
let wrapper;

describe('Render the Nav Bar', () => {
  it('Should render successfully', async () => {
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    sinon.stub(window.location, 'replace');
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});

describe('Render the Nav Bar', () => {
  beforeAll(() => {
    global.window = Object.create(window);
    url = '?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MiwiZW1haWwiOiJlcmljLm1hbGFiYTZAZ21haWwuY29tIiwicm9sZSI6InVzZXIiLCJ1c2VybmFtZSI6IkVyaWM2IiwiaWF0IjoxNTcxMjA1MTYyLCJleHAiOjE1NzEyOTE1NjJ9.ZdMucW9PC4K9PLJKX1G7cnWyHA2ihGfcBqhRT1-q4Co#';
    Object.defineProperty(window, 'location', {
      value: { href: url },
      writable: true,

    });
  });
  it('Should render successfully', async () => {
    expect(window.location.href).toEqual(url);
  });
});
