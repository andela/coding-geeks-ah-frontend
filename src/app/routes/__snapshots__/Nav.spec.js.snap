// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the Nav Bar Should render successfully 1`] = `ReactWrapper {}`;
